flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/api/bridge/java/StreamTableEnvironment.java

```diff
@@ -122,56 +118,6 @@ static StreamTableEnvironment create(
         return StreamTableEnvironmentImpl.create(executionEnvironment, settings);
     }
 
-    /**
-     * Registers a {@link TableFunction} under a unique name in the TableEnvironment's catalog.
-     * Registered functions can be referenced in Table API and SQL queries.
-     *
-     * @param name The name under which the function is registered.
-     * @param tableFunction The TableFunction to register.
-     * @param <T> The type of the output row.
-     * @deprecated Use {@link #createTemporarySystemFunction(String, UserDefinedFunction)} instead.
-     *     Please note that the new method also uses the new type system and reflective extraction
-     *     logic. It might be necessary to update the function implementation as well. See the
```

Comment:
I notice there are other deprecated methods in this file that have not been removed. Is there a reason to not remove those also?

Reply:
The scope of this subtask is limited to the removal of `registerFunction` method in this class.